geomstats/geometry/general_linear.py

```diff
@@ -1,192 +1,125 @@
+        The orbit is returned as the path function satisfying:
 
         Parameters
         ----------
-        tangent_vec
-        base_point
-        point_type
+        point : array-like, shape=[n, n]
+            Target point.
+        base_point : array-like, shape=[n, n], optional
+            Base point. Defaults to identity.
 
         Returns
         -------
-        group_exp
-        """
-        tangent_vec = gs.to_ndarray(tangent_vec, to_ndim=3)
-        base_point = gs.to_ndarray(base_point, to_ndim=3)
+        path : callable
+            The one-parameter orbit.
+            Satisfies `path(0) = base_point` and `point(1) = point`.
 
-        tangent_vec_at_identity = self.compose(
-            self.inverse(base_point), tangent_vec)
+        Note
+        ----
+        Denoting `point` by :math: `g` and `base_point` by :math: `h`,
+        the orbit :math: `\gamma` satisfies:
 
-        exp_from_identity = self.exp_from_identity(
-                tangent_vec_at_identity)
+        .. math::
 
-        group_exp = self.compose(
-            base_point, exp_from_identity)
+            \gamma(t) = {\mathrm e}^{t X} \cdot h \\
+            \quad {\mathrm with} \quad\\
+            {\mathrm e}^{X} = g h^{-1}
 
-        return group_exp
+        The path is not uniquely defined and depends on
+        the choice of :math: `V` returned by `class.log`.
 
-    def log_from_identity(self, point, point_type=None):
-        """Compute group logarithm at the identity.
+        Vectorization
+        -------------
+        Return a collection of trajectories (4-D array)
+        from a collection of input matrices (3-D array).
 
-        Group logarithm of the Lie group of
-        all invertible matrices at the identity.
-
-        Parameters
-        ----------
-        point
-        point_type
-
-        Returns
-        -------
-        group_log
+        Will work when expm gets properly 4-D vectorized.
         """
-        point = gs.to_ndarray(point, to_ndim=3)
-        group_log = gs.linalg.logm(point)
-
-        return gs.real(group_log)
-
-    def log_not_from_identity(
-            self, point, base_point, point_type=None):
-        """Compute group logarithm at a base point.
-
-        Group logarithm of the Lie group of all invertible matrices.
-
-        Parameters
-        ----------
-        point
-        base_point
-        point_type
-
-        Returns
-        -------
-        group_log
-        """
-        point = gs.to_ndarray(point, to_ndim=3)
-        base_point = gs.to_ndarray(base_point, to_ndim=3)
-
-        point_near_identity = self.compose(
-            self.inverse(base_point), point)
-
-        log_from_identity = self.log_from_identity(
-            point_near_identity)
+        vector = cls.log(point, base_point)
 
-        group_log = self.compose(
-            base_point, log_from_identity)
+        def path(time):
+            vecs = gs.einsum('t,...ij->...tij', time, vector)
```

Comment:
We need to check if ellipsis work in tf and pytorch (later PR probably)

Reply:
on this subject, I looked for the pytorch einsum docs which seem ok:
 https://pytorch.org/docs/stable/torch.html#torch.einsum
there's definitely some testing to do!